Source/items.cpp

```diff
@@ -1445,7 +1459,7 @@ int RndUItem(MonsterStruct *monster)
 		}
 	}
 
-	return ril[GenerateRnd(ri)];
+	return ril[GenerateNonNegativeRnd(ri)];

```

Comment:
This seemed like the safest way to handle this, I don't think reading invalid memory was actually useful for any generated items but it may break compatibility. That said, with the way it was working previously any unrelated code change potentially could've broken compatibility, no way to know without tests for the exceptional case.